### PR TITLE
build(deps): bump craft-archives

### PR DESCRIPTION
**Pull request by tigarmo**
This update fixes LP#2083013, where some package-repositories declarations could conflict with default sources present in Noble.

- [ ] Have you followed the [guidelines for contributing](https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox run -m lint`?
- [ ] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: requirements-devel.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ codespell==2.3.0
 colorama==0.4.6
 coverage==7.6.1
 craft-application==4.2.7
-craft-archives==2.0.0
+craft-archives @ git+https://github.com/canonical/craft-archives@work/CRAFT-3538-noble-key-conflict
```
Comment (lengau) [inline requirements-devel.txt:28]: Just putting this here to make it easy if you take my suggestion about releasing today.

```suggestion
craft-archives==2.0.1
```

File: requirements-docs.txt [hunk 2]
```diff
@@ -20,7 +20,7 @@ charset-normalizer==3.3.2
 click==8.1.7
 colorama==0.4.6
 craft-application==4.2.7
-craft-archives==2.0.0
+craft-archives @ git+https://github.com/canonical/craft-archives@work/CRAFT-3538-noble-key-conflict
```
Comment (lengau) [inline requirements-docs.txt:23]: Just putting this here to make it easy if you take my suggestion about releasing today.

```suggestion
craft-archives==2.0.1
```

File: requirements.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@ chardet==5.2.0
 charset-normalizer==3.3.2
 click==8.1.7
 craft-application==4.2.7
-craft-archives==2.0.0
+craft-archives @ git+https://github.com/canonical/craft-archives@work/CRAFT-3538-noble-key-conflict
```
Comment (lengau) [inline requirements.txt:11]: Just putting this here to make it easy if you take my suggestion about releasing today.

```suggestion
craft-archives==2.0.1
```

File: tests/spread/core24/package-repositories/task.yaml [hunk 4]
```diff
@@ -9,6 +9,7 @@ environment:
   SNAP/test_multi_keys: test-multi-keys
   SNAP/test_foreign_armhf: test-foreign-armhf
   SNAP/test_foreign_i386: test-foreign-i386
+  SNAP/test_key_conflict: test-key-conflict
```
Comment (lengau) [inline tests/spread/core24/package-repositories/task.yaml:12]: This is an excellent example of well-designed tests sight here.

File: tests/spread/core24/package-repositories/test-key-conflict/snapcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+name: test-key-conflict
+version: '1.0'
+summary: test package repos with keys already present on the system
```
Comment (lengau) [inline tests/spread/core24/package-repositories/test-key-conflict/snapcraft.yaml:3]: Like the well-designed tests above, this is a well-written example that makes it really clear. Thanks!

File: tests/spread/core24/package-repositories/test-key-conflict/snapcraft.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+confinement: strict
+base: core24
+platforms:
+  amd64:
+
+package-repositories:
+  # In core24 this pair of url + suite is already listed in
+  # /etc/apt/sources.list.d/ubuntu.sources. Add it here to make sure this setup
+  # is working.
+  - type: apt
```
Comment (lengau) [inline tests/spread/core24/package-repositories/test-key-conflict/snapcraft.yaml:14]: As with my comment on craft-archives, is it worth putting an Ubuntu Ports repo in here and seeing if we can stage something from one of those platforms?

**Review by lengau — APPROVED** [review]

File: tests/spread/core24/package-repositories/test-key-conflict/snapcraft.yaml (hunk 6 above)
Comment (lengau) [reply tests/spread/core24/package-repositories/test-key-conflict/snapcraft.yaml:14]: Tiago explained separately that this wouldn't cause an issue.

**Review by mr-cal — APPROVED** [review]
Approving with the assumption that @tigarmo will update the requirement files after releasing craft-archives.

**Pull request merged**